lib/merit/controller_extensions.rb

```diff
@@ -46,16 +46,20 @@ def had_errors?
       target_object.respond_to?(:errors) && target_object.errors.try(:present?)
     end
 
-    def target_object
-      target_obj = instance_variable_get(:"@#{controller_name.singularize}")
-      if target_obj.nil?
+    def warn_no_object_found
         str = '[merit] No object found, you might need a ' \
           "'@#{controller_name.singularize}' variable in " \
           "'#{controller_path}_controller' if no reputation is applied. " \
           'If you are using `model_name` option in the rule this is ok.'
         Rails.logger.warn str
+    end
+
+    def target_object
+      variable_name = :"@#{controller_name.singularize}"
+      if instance_variable_defined?(variable_name)
+        target_obj = instance_variable_get(variable_name)
       end
-      target_obj
+      target_obj.nil? ? warn_no_object_found : target_obj
```

Comment:
In the case where `target_obj.nil?` this new code returns `true`, because that's what `Rails.logger.warn`. It should instead return `nil`, either explicit or implicit (no `else` branch in the `if`).